apps/src/lib/levelbuilder/lesson-editor/ResourcesEditor.jsx

```diff
@@ -103,9 +106,9 @@ export default class ResourcesEditor extends Component {
     });
   };
 
-  addResource = e => {
+  addResource = resource => {
     var {resources} = this.state;
-    resources.push(e.resource);
+    resources.push(resource);
```

Comment:
I could be wrong but I think you want to avoid directly changing the resources here by pushing. Instead making a copy of the resources, adding to that, and then updating with setState

Reply:
Showing off my mediocre React/JS knowledge here -- how would that be different than what's here? This does use `setState` to update the resource.

Reply:
This post gives some background. the problem is that you are mutating `this.state`: https://jkettmann.com/how-to-accidentally-mutate-state-and-why-not-to/ because `push` modifies the array. a simple fix would be to do `resources = resources.concat(resource)` instead, which returns a separate copy.

Reply:
Thanks!